loki/dimension.py

```diff
@@ -69,68 +81,90 @@ def variables(self):
         return (self.index, self.size) + self.bounds
 
     @property
-    def size(self):
+    def sizes(self):
         """
-        String that matches the size expression of a data space (variable allocation).
+        Tuple of strings that matche the primary size and all secondary size expressions.
```

Comment:
For this description to be true, don't we also need to add the output of `size_expressions`?

Reply:
No, I don't think we do. `self._size` already contains all expressions that describe the size, including aliases. The only thing that `size_expressions` would add are expressions that are derived from iteration space bounds. Keeping the two explicitly separate is an intended outcome of this PR. It might, however, make sense to use `self.sizes` in `size_expressions` instead of only the primary size expression

Reply:
Yes, IIRC in some corner case the difference between "secondary alias" and "derived" is significant, so the two stay distinct. Once we mop up the aliases in the regression tests, we can mop this up though, I think (I'd need to check which one exactly though). In any case, I'll add a note to the docstring to clarify.